anna/my-app/src/components/sortBox/SortBox.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC, useEffect, useState } from 'react';
+import Container from '../container';
+import Button from '../button';
+import './sortBox.css';
+import { useLocation } from 'react-router-dom';
+import * as QueryString from "query-string";
+
+interface ISortBoxProps {
+  movieCount?: number
+  movieGanre?: string
+  onSortByChange?: (arg0: SortType)=> void;
+  sortBy?: SortType
+}
+
+export enum SortType {
+  ReleaseDate = "release_date",
+  Rating = "vote_average"
+}
+
+const SortBox: FC<ISortBoxProps> = (props) => {
+  const location = useLocation();
+  const searchParams = QueryString.parse(location.search);  
+
+  const [sort, setSort] = useState(searchParams.sortBy as string);
```

Comment:
seems like you don't need sort in state. You have sort in Home page. It's a bad practice to manage same state in several components. Use sort from props